### ISSUE TITLE
Is there a hook for 'before_feature' in pytest-bdd?

### ISSUE THREAD

**Issue by albertnjoku** [pytest-dev/pytest-bdd#688]
Is there a way I can execute a code based on a tag inserted at the feature level in my feature file? I have the same working at the scenario level but the requirement is to run once for all scenarios in the feature file based on tag

**Comment by Jiff21** [comment]
You can use background

```
Background: This sets up before all the scenarios
        Given I do some stuff
        And I do other stuff
```

You can use also something like this to run after all tests.
```
@pytest.fixture(scope="module", autouse=True)
def after_module():
    # Run Before Feature code here
    print(this runs before)
    yield True
    # Runs after the module
    print(this runs after)
```

Of create a fixture in indiviual test.py step files so you can attach it to `@scenario('../FeatureFiles/Test.feature', 'Test Fixture'). 

But what I can't figure out a way is an after_feature hook because I'm using `scenarios('features')` to import all my features then the fixture only runs before and after all features.

**Comment by youtux** [comment]
pytest-bdd tags are just really pytest markers. Refer to the pytest documentation on how to run code depending on tags applied to a test item:  https://docs.pytest.org/en/8.3.x/how-to/fixtures.html#using-markers-to-pass-data-to-fixtures

TL;DR put this in your test files / `conftest.py`:

```python

@pytest.fixture(autouse=True)
def run_my_code_if_tagged(request):
    tagged = request.node.get_closest_marker("my_tag") is not None
    if tagged:
        # run my custom code
```

**Comment by Jiff21** [comment]
This doesn't work as an after all hook when using `scenarios('features')`.  `@pytest.fixture(scope="module", autouse=True)
` will run after all Feature files. `@pytest.fixture(scope="class", autouse=True)` will run before and after all scenarios.
